pubsub.go

```diff
@@ -373,15 +413,23 @@ func (p *PubsubValueStore) Cancel(name string) (bool, error) {
 	return ok, nil
 }
 
-func (p *PubsubValueStore) handleSubscription(ctx context.Context, sub *pubsub.Subscription, key string, cancel func()) {
-	defer sub.Cancel()
+func (p *PubsubValueStore) handleSubscription(ctx context.Context, ti *topicInfo, key string, cancel func()) {
+	defer func() {
+		if err := ti.topic.Close(); err != nil {
+			if err != context.Canceled && err != context.DeadlineExceeded {
+				log.Error(err)
+			}
+		}
+	}()
+	defer ti.evts.Cancel()
+	defer ti.sub.Cancel()
```

Comment:
Do we need to explicitly close these?

Reply:
It's probably not a big deal the ways this function exits are:
1. The PubSubValueStore's context is cancelled
2. The internal subscription was manually cancelled

Given that the second shouldn't occur outside of this code (and if it does it's probably unsafe) and the first means the whole service is dead anyway it's probably fine to leave it. Was keeping it mostly for consistency with the previous system, although probably having an internal `Close` function as in https://github.com/libp2p/go-libp2p-pubsub-router/issues/35 is probably better.

I can look into it.

Reply:
The cancels are idempotent so it shouldn't be a big deal, also I took a stab at fixing #35 as part of cleaning this up. lmk what you think.